apps/aecore/src/txs/aec_tx.erl

```diff
@@ -40,29 +40,57 @@ fee(#spend_tx{fee = F}) ->
     F.
 
 -spec apply_signed(list(signed_tx()), trees(), non_neg_integer()) ->
-                          {ok, trees()} | {error, term()}.
-apply_signed([], Trees, _Height) ->
-    {ok, Trees};
-apply_signed([SignedTx | Rest], Trees0, Height) ->
-    case aec_tx_sign:verify(SignedTx) of
-        ok ->
-            Tx = aec_tx_sign:data(SignedTx),
-            case check_single(Tx, Trees0, Height) of
-                {ok, Trees1} ->
-                    {ok, Trees2} = process_single(Tx, Trees1, Height),
-                    apply_signed(Rest, Trees2, Height);
-                {error, _Reason} = Error ->
-                    Error
-            end;
-        {error, _Reason} = Error ->
-            Error
-    end.
+                          {ok, trees()}.
+apply_signed(SignedTxs, Trees0, Height) ->
+    Txs = verify_and_extract_txs(SignedTxs),
```

Comment:
As function `verify_and_extract_txs` silently discards txs, the coinbase tx may be silently discarded too!
Please make a proposal where in the code there is the hardcoded expectation that the first tx among the txs being included in the block to be mined is the coinbase tx (so that the reader does not need to assume so with the risk that block w/o coinbase transaction is mined and then rejected by other nodes in network).

Reply:
There is a check on coinbase tx coint in https://github.com/aeternity/epoch/blob/master/apps/aecore/src/aec_blocks.erl#L246 (it is executed when accepting a block from the network), so we may explicitly expect 1 coinbase tx there, wdyt?

Reply:
> There is a check on coinbase tx coint in https://github.com/aeternity/epoch/blob/master/apps/aecore/src/aec_blocks.erl#L246 (it is executed when accepting a block from the network), so we may explicitly expect 1 coinbase tx there, wdyt?

If that function is not called on genesis (I did not check, seems out of scope for this PR), check of equal to 1 is ok.

Though wrt mining, I did not see validation of coinbase tx on block candidate (please let me know if I missed it).